lib/Zonemaster/CLI/TestCaseSet.pm

```diff
@@ -0,0 +1,233 @@
+package Zonemaster::CLI::TestCaseSet;
+use 5.014;
+use warnings;
+use utf8;
+
+use Carp qw( croak );
+
+=head1 NAME
+
+    Zonemaster::CLI::TestCaseSet - A mutable set of test case names.
+
+=head1 SYNOPSIS
+
+    use Zonemaster::CLI::TestCaseSet;
+
+    # Construct a working subset of test cases {alpha01, alpha02, alpha03,
+    # beta01} of test cases out of the full set {alpha01, alpha02, alpha03,
+    # beta01, beta02} distributed across the test modules {alpha, beta}.
+    my $working_set = Zonemaster::CLI::TestCaseSet->new(
+        \qw( alpha01 alpha02 alpha03 beta01 ),
+        {
+            alpha => \qw( alpha01 alpha02 alpha03 ),
+            beta  => \qw( beta01 beta02 ),
+        },
+    );
+
+    # Parse a modifier expression into a list of modifiers.
+    my @modifiers = Zonemaster::CLI::TestCaseSet->parse_modifier_expr( '-alpha+alpha02' );
+
+    # Traverse the list of modifiers, chunked into (operator, term) pairs.
+    while ( @modifiers ) {
+        my $op   = shift @modifiers;
+        my $term = shift @modifiers;
+
+        # Modify the working subset by applying each operator and term.
+        if ( !$working_set->apply_modifier( $op, $term ) ) {
+            die "Error: Unrecognized term '$term'.\n";
+        }
+    }
+
+    # Make sure the working subset ends up in the expected state.
+    if ( join(' ', $working_set->to_list) ne 'alpha02 beta01' ) {
+        die;
+    }
+
+=head1 DESCRIPTION
+
+A TestCaseSet primarily represents an immutable full set of test cases and a
+mutable subset thereof. The full set of test cases is distributed across the
+set of test modules.
+
+=head2 TERM EXPANSION
+
+Terms are expanded in one of three ways.
+
+=over 4
+
+=item The full set of test cases.
```

Comment:
```suggestion
=item The full set of test cases in the default profile.
```

Reply:
As I replied before, the TestCaseSet doesn't have any concept of profiles at all. It just works with whatever you tell it in the constructor.

Reply:
When you select "all" you get all test cases in the default profile.

Could you give a better explanation and description of `TestCaseSet.pm`? I do not understand.

Reply:
> When you select "all" you get all test cases in the default profile.

This true if and only if you provide "all test cases in the default profile" to the TestCaseSet constructor. In other words, it depends on how the module is used. The documentation of a module should not depend on how that module is being used.

Reply:
Can you please give a more explicit description of the module? I still do not understand it.